pkg/config/global.go

```diff
@@ -31,87 +30,54 @@ const (
 )
 
 var (
-	policyRepoPath       = os.Getenv("HOME") + "/.terrascan"
-	policyBasePath       = policyRepoPath + "/pkg/policies/opa/rego"
-	errTomlKeyNotPresent = fmt.Errorf("%s key not present in toml config", policyConfigKey)
+	policyRepoPath = os.Getenv("HOME") + "/.terrascan"
+	policyBasePath = policyRepoPath + "/pkg/policies/opa/rego"
 )
 
 func init() {
 	// If the user specifies a config file in TERRASCAN_CONFIG,
 	// overwrite the defaults with the values from that file.
 	// Retain the defaults for members not specified in the file.
-	LoadGlobalConfig(os.Getenv(configEnvvarName))
+	if err := LoadGlobalConfig(os.Getenv(configEnvvarName)); err != nil {
+		zap.S().Error("error while loading global config", zap.Error(err))
+	}
 }
 
 // LoadGlobalConfig loads policy configuration from specified configFile
 // into var Global.Policy.  Members of Global.Policy that are not specified
 // in configFile will get default values
-func LoadGlobalConfig(configFile string) {
+func LoadGlobalConfig(configFile string) error {
 	// Start with the defaults
-	Global.Policy = PolicyConfig{
+	Global.Policy = Policy{
 		BasePath: policyBasePath,
 		RepoPath: policyRepoPath,
 		RepoURL:  policyRepoURL,
 		Branch:   policyBranch,
 	}
 
-	if len(configFile) > 0 {
-		p, err := loadConfigFile(configFile)
-		if err != nil {
-			zap.S().Error(err)
-			return
-		}
-		if len(p.Policy.BasePath) > 0 {
-			Global.Policy.BasePath = p.Policy.BasePath
-		}
-		if len(p.Policy.RepoPath) > 0 {
-			Global.Policy.RepoPath = p.Policy.RepoPath
-		}
-		if len(p.Policy.RepoURL) > 0 {
-			Global.Policy.RepoURL = p.Policy.RepoURL
-		}
-		if len(p.Policy.Branch) > 0 {
-			Global.Policy.Branch = p.Policy.Branch
-		}
+	if configFile == "" {
+		zap.S().Debug("global config env variable is not specified")
+		return nil
 	}
-}
-
-func loadConfigFile(configFile string) (GlobalConfig, error) {
-	p := GlobalConfig{}
 
-	config, err := LoadConfig(configFile)
+	configReader, err := NewTerrascanConfigReader(configFile)
 	if err != nil {
-		return p, ErrNotPresent
+		return err
 	}
 
-	keyConfig := config.Get(policyConfigKey)
-	if keyConfig == nil {
-		return p, errTomlKeyNotPresent
+	if len(configReader.GetPolicyConfig().BasePath) > 0 {
```

Comment:
I think we shouldn't do these length checks. The reason is, if the global config happens to contain an existing config, running loadConfigFile could leave the user with a partially loaded config file (values from old config + some values from new config), and that is misleading to the user. I would expect if the new config file has an empty path, an empty path should be loaded. That is an error the user would have to fix. I would just remove the length check and set it directly: Global.Policy.BasePath = configReader.GetPolicyConfig().BasePath, or simply assign the entire config in one shot.